core/txpool/txpool.go

```diff
@@ -2303,10 +2303,19 @@ func (pool *TxPool) demoteUnexecutables() {
 			pool.enqueueTx(hash, tx, false, false)
 		}
 
-		pendingGauge.Dec(int64(oldsLen + dropsLen + invalidsLen))
+		// Drop all transactions that no longer have valid TxOptions
+		txConditionalsRemoved := list.FilterTxConditional(pool.currentState)
+
+		for _, tx := range txConditionalsRemoved {
+			hash := tx.Hash()
+			log.Trace("Removed invalid conditional transaction", "hash", hash)
```

Comment:
can we `log.Trace` after the tx has been actually removed?

Reply:
[Done](https://github.com/maticnetwork/bor/pull/945/commits/abe87af0bb99b7995b621af5eb41d50b4a964e0f).